instrumentation-genai/opentelemetry-instrumentation-google-genai/src/opentelemetry/instrumentation/google_genai/generate_content.py

```diff
@@ -129,21 +131,60 @@ def _determine_genai_system(models_object: Union[Models, AsyncModels]):
     return _get_gemini_system_name()
 
 
-def _get_config_property(
-    config: Optional[GenerateContentConfigOrDict], path: str
-) -> Any:
+def _to_dict(value: object):
+    if isinstance(value, dict):
+        return value
+    if hasattr(value, "model_dump"):
+        return value.model_dump()
+    return json.loads(json.dumps(value))
+
+
+def _add_request_options_to_span(
+    span, config: Optional[GenerateContentConfigOrDict]
+):
     if config is None:
-        return None
-    path_segments = path.split(".")
-    current_context: Any = config
-    for path_segment in path_segments:
-        if current_context is None:
-            return None
-        if isinstance(current_context, dict):
-            current_context = current_context.get(path_segment)
-        else:
-            current_context = getattr(current_context, path_segment)
-    return current_context
+        return
+    span_context = span.get_span_context()
+    if not span_context.trace_flags.sampled:
+        # Avoid potentially costly traversal of config
+        # options if the span will be dropped, anyway.
+        return
+    # Automatically derive attributes from the contents of the
+    # config object. This ensures that all relevant parameters
+    # are captured in the telemetry data (except for those
+    # that are excluded via "exclude_keys").
+    attributes = flatten_dict(
+        _to_dict(config),
+        # A custom prefix is used, because the names/structure of the
+        # configuration is likely to be specific to Google Gen AI SDK.
+        key_prefix=CUSTOM_LLM_REQUEST_PREFIX,
+        exclude_keys=[
+            # System instruction can be overly long for a span attribute.
+            # Additionally, it is recorded as an event (log), instead.
+            "gen_ai.gcp.request.system_instruction",
+            # Headers could include sensitive information, therefore it is
+            # best that we not record these options.
+            "gen_ai.gcp.request.http_options.headers",
+        ],
+        # Although a custom prefix is used by default, some of the attributes
+        # are captured in common, standard, Semantic Conventions. For the
+        # well-known properties whose values align with Semantic Conventions,
+        # we ensure that the key name matches the standard SemConv name.
+        rename_keys={
+            # TODO: add more entries here as more semantic conventions are
+            # generalized to cover more of the available config options.
+            "gen_ai.gcp.request.temperature": "gen_ai.request.temperature",
+            "gen_ai.gcp.request.top_k": "gen_ai.request.top_k",
```

Comment:
For the RHS standard semconv keys, we should use the constants from the semconv package

Reply:
Marking resolved.